src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Emitter.cs

```diff
@@ -215,36 +227,43 @@ private bool EmitObjectInit(TypeSpec type, string memberAccessExpr, Initializati
                 return true;
             }
 
-            private void EmitCastToIConfigurationSection()
+            private void EmitInterceptsLocationAttrDecl()
             {
-                string sectionTypeDisplayString;
-                string exceptionTypeDisplayString;
-                if (_useFullyQualifiedNames)
-                {
-                    sectionTypeDisplayString = "global::Microsoft.Extensions.Configuration.IConfigurationSection";
-                    exceptionTypeDisplayString = FullyQualifiedDisplayString.InvalidOperationException;
-                }
-                else
-                {
-                    sectionTypeDisplayString = Identifier.IConfigurationSection;
-                    exceptionTypeDisplayString = nameof(InvalidOperationException);
-                }
-
+                _writer.WriteLine();
                 _writer.WriteLine($$"""
-                    if ({{Identifier.configuration}} is not {{sectionTypeDisplayString}} {{Identifier.section}})
+                    namespace System.Runtime.CompilerServices
                     {
-                        throw new {{exceptionTypeDisplayString}}();
+                        using System;
+                        using System.CodeDom.Compiler;
+
+                        {{Expression.GeneratedCodeAnnotation}}
+                        [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+                        file sealed class InterceptsLocationAttribute : Attribute
+                        {
+                            public InterceptsLocationAttribute(string filePath, int line, int column)
```

Comment:
Did we want to add something like https://github.com/dotnet/sdk/blob/4249701de6f005c670c7f371141e772d02f89156/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets#L64 to our NuGet package targets, or will we ask users to set this?  @eerhardt @tarekgh @captainsafia

Reply:
I think we decided to mimic whatever the aspnet is doing in the web project. CC @ericstj

Reply:
ASP.NET doesn't ship a nuget package so we're special here.  There's an ongoing discussion around this to decide what we want to do.